src/x11/x.c

```diff
@@ -703,6 +703,14 @@ static void x_win_move(int width, int height)
                 y = scr->dim.y + xctx.geometry.y;
         }
 
+        if (xctx.geometry.center && xctx.geometry.x == 0) {
+                x = (scr->dim.x + (scr->dim.w - width) / 2);
+        }
+
+        if (xctx.geometry.center && xctx.geometry.y == 0) {
+                y = (scr->dim.y + (scr->dim.y - height) / 2);
```

Comment:
You probably meant `scr->dim.h` on the average calculation here, `y` is the screen position and `h` the height.